test/unit/test_runner.py

```diff
@@ -152,3 +144,32 @@ def test_status_callback_interface(rc, mocker):
     assert runner.status_handler.call_count == 1
     runner.status_handler.assert_called_with(dict(status='running', runner_ident=str(rc.ident)), runner_config=runner.config)
     assert runner.status == 'running'
+
+
+@pytest.mark.parametrize('runner_mode', ['pexpect', 'subprocess'])
+def test_stdout_file_write(rc, runner_mode):
+    rc.command = ['echo', 'hello_world_marker']
+    rc.runner_mode = runner_mode
+    status, exitcode = Runner(config=rc).run()
+    assert status == 'successful'
+    stdout_path = Path(rc.artifact_dir) / 'stdout'
+    # this can be subject to a race condition so we will be patient with the check
+    start_time = time.time()
+    while time.time() - start_time < 2.0:
```

Comment:
Use `iterate_timeout()` here from `test/utils/common.py`. You can find examples in the test code.